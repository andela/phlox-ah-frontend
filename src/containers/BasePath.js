export * from '../requests/LoginRequests';
export * from '../util/AsyncUtil';
export * from '../actionTypes/UserConstants';
export * from '../actions/MsgInfoActions';
export * from '../components/Input/Input';
export * from '../components/Button/Button';
export * from '../requests/SignupRequests';
export * from '../requests/ArticleRequests';
export * from '../requests/ProfileRequest';

