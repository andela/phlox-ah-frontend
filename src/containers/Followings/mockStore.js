export const followerStore = {
  loading: false,
  success: false,
  followUser: {
    followings: [{
      id: 1,
      username: 'johndoe',
      email: 'jd@something.com',
      Profile: {
        firstName: 'John',
        lastName: 'Doe',
        profileImage: 'https://res.cloudinary.com/sammykeyz/image/upload/v1542924983/phlox/1542924981230_me.jpg',
        gender: 'male',
        contact: 'Test street',
        bio: 'Lowkey bad guy'
      }
    }],
    followers: [{
      id: 1,
      username: 'johndoe',
      email: 'jd@something.com',
      Profile: {
        firstNa: 'John',
        lastName: 'Doe',
        profileImage: 'https://res.cloudinary.com/sammykeyz/image/upload/v1542924983/phlox/1542924981230_me.jpg',
        gender: 'male',
        contact: 'Test street',
        bio: 'Lowkey bad guy'
      }
    }],
  }
};
