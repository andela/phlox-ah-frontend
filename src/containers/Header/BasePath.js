export * from '../../actionTypes/UserConstants';
export * from '../../util/AsyncUtil';
