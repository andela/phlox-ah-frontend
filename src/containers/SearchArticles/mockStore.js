export const articleStore = {
  search: {
    loading: true,
    success: true,
    failure: false,
    searchResult: [
      {
        id: 3,
        title: 'Second article by nelso',
        body: 'The second article by Nelson v Second article by Nelson.The second article by Nelson.<p><br></p>',
        slug: 'Second-article-by-nelson-09afb545-4110-4656-8378-c6b6f192eb92',
        description: 'Description of second article by nelson',
        imgUrl: 'https://res.cloudinary.com/dafcdw02e/image/upload/v1542892294/phlox/1542892286288_adult-agent-approval-684385.jpg',
        readTime: 1,
        ratingAverage: 0,
        ArticlesTags: {
          articleId: 3,
          tagId: 1,
          createdAt: '2018-11-22T13:11:20.097Z',
          updatedAt: '2018-11-22T13:11:20.097Z'
        }
      }
    ],
    error: null
  }
};
