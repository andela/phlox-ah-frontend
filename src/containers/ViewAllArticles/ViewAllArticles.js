import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-materialize';

import { ArticleCard } from '../../components/ArticleCard/ArticleCard';
import Sidebar from '../Sidebar/Sidebar';
import { getArticles } from '../../requests/ArticleRequests';
import './ViewAllArticles.scss';


/**
 * @class ViewAllArticles
 * @extends {Component}
 */
class ViewAllArticles extends Component {
  /**
   *Creates an instance of Home.
   * @param {object} props
   * @memberof ViewAllArticles
   */
  constructor(props) {
    super();
    this.state = {
      success: false,
      articles: []
    };
  }

  /**
   * @description - This method runs whenever redux state changes
   * @returns {object} state
   * @param {object} props
   */
  static getDerivedStateFromProps(props) {
    return props;
  }

  /**
   * @returns {func} tag
   */
  componentDidMount() {
    this.props.getArticles();
  }


  /**
   * @description - This method shows all articles
   * @returns {jsx} - jsx
   */
  showAllArticles() {
    const { articles } = this.state;
    return articles.map((article) => {
      const username = article.User ? article.User.username : '';
      return (
        <Col s={12} m={12} l={12} xl={6} key={article.id}>
          <ArticleCard
            size="medium"
            pic={article.imgUrl}
            title={article.title}
            description={article.description}
            createdAt={article.createdAt}
            author={username}
            slug={article.slug}
          />
        </Col>);
    });
  }

  /**
   *
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   */
  render() {
    const { success } = this.state;
    return (
      <div>
         <main>
          <div className="site-container">
          <Row>
              <Col s={12} m={12} l={8} xl={9} className="column-1">
                <Row>
                  { success ? this.showAllArticles() : <h5 className="center no-tags">There are no articles at the moment</h5> }
                </Row>
              </Col>
              <Col s={12} m={9} l={4} xl={3} className="sidebar">
                <Sidebar/>
              </Col>
          </Row>
          </div>
        </main>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  articles: state.article.articles,
  success: state.article.success
});

ViewAllArticles.propTypes = {
  articles: PropTypes.array,
  getArticles: PropTypes.func,
  match: PropTypes.object,
  success: PropTypes.bool
};

export default connect(mapStateToProps, { getArticles })(ViewAllArticles);
