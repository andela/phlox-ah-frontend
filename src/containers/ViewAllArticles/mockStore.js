export const articleStore = {
  article: {
    loading: true,
    success: true,
    failure: false,
    articles: [
      {
        id: 3,
        title: 'Second article by nelson',
        body: 'The second article by Nelson v Second article by Nelson.The second article by Nelson.<p><br></p>',
        slug: 'Second-article-by-nelson-09afb545-4110-4656-8378-c6b6f192eb92',
        description: 'Description of second article by nelson',
        imgUrl: 'https://res.cloudinary.com/dafcdw02e/image/upload/v1542892294/phlox/1542892286288_adult-agent-approval-684385.jpg',
        readTime: 1,
        userId: 2,
        categoryId: 1,
        ratingAverage: 0,
        featured: 0,
        status: 'published',
        createdAt: '2018-11-22T13:11:20.033Z',
        updatedAt: '2018-11-22T13:11:44.537Z',
        Tags: [
          {
            id: 1,
            name: 'fake news',
            createdAt: '2018-11-22T13:11:19.925Z',
            updatedAt: '2018-11-22T13:11:19.925Z',
            ArticlesTags: {
              articleId: 3,
              tagId: 1,
              createdAt: '2018-11-22T13:11:20.097Z',
              updatedAt: '2018-11-22T13:11:20.097Z'
            }
          }
        ],
        Category: {
          id: 1,
          category: 'Technology',
          createdAt: '2018-11-22T23:00:00.000Z',
          updatedAt: '2018-11-22T23:00:00.000Z'
        },
        User: {
          username: 'nelson',
          Profile: null
        },
        likes: [
          {
            id: 3,
            like: null,
            createdAt: '2018-11-22T13:12:23.783Z',
            updatedAt: '2018-11-22T13:17:11.857Z',
            articleSlug: 'Second-article-by-nelson-09afb545-4110-4656-8378-c6b6f192eb92',
            userId: 2,
            User: {
              username: 'nelson',
              email: 'nelson.nnaji@andela.com'
            }
          }
        ]
      },
      {
        id: 2,
        title: 'new test article by udochukwu',
        body: '<p>This enables you to deliver on time regardless of unforeseen blockers. Products are built on timelines. Sometimes these timelines are based on launch dates, financial projections, product roadmaps, strategic objectives .etc. Every task done and feature implemented contributes to meeting the timelines set for the overall product. In addition, features have dependencies and timelines on one feature contribute to timelines on others. If you don’t learn how to estimate timelines for delivery, it will translate to you making promises you can’t keep. Risks you didn’t foresee will always creep up and mess up your delivery timelines. Mastering the ability to estimate accurately enables you to build a great relationship with your clients /partners and to reinforce your brand as a professional developer who can be trusted.</p>\n\n<p>Every task you pick up on a team project will require you to scope it out, assign it a weight based on its complexity and estimate how long it will take you to complete it. Every bug you encounter will be a potential blocker to meeting your estimated timeline. Every new concept or technology that you need to learn on the path to implementing a feature is a potential hindrance to meeting estimated timelines. Software development is problem-solving on the clock. If you fail to master the ability to foresee and account for risks that arise in the process of building a product, you will struggle to estimate properly and constantly fail to deliver on time. You will basically be a dev who ‘talks a lot, doesn’t deliver!’</p>',
        slug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
        description: 'Description here and here too',
        imgUrl: 'https://images.pexels.com/photos/179912/pexels-photo-179912.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        readTime: 1,
        userId: 1,
        categoryId: 1,
        ratingAverage: 4,
        featured: 0,
        status: 'published',
        createdAt: '2018-11-20T08:00:03.743Z',
        updatedAt: '2018-11-21T14:07:53.239Z',
        Tags: [],
        Category: {
          id: 1,
          category: 'Technology',
          createdAt: '2018-11-22T23:00:00.000Z',
          updatedAt: '2018-11-22T23:00:00.000Z'
        },
        User: {
          username: 'nnaji',
          Profile: null
        },
        likes: [
          {
            id: 2,
            like: false,
            createdAt: '2018-11-20T11:23:02.778Z',
            updatedAt: '2018-11-22T12:31:07.855Z',
            articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
            userId: 2,
            User: {
              username: 'nelson',
              email: 'nelson.nnaji@andela.com'
            }
          },
          {
            id: 1,
            like: true,
            createdAt: '2018-11-20T10:47:12.913Z',
            updatedAt: '2018-11-20T11:13:16.155Z',
            articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
            userId: 1,
            User: {
              username: 'nnaji',
              email: 'nnaji_udochukwu@yahoo.com'
            }
          }
        ]
      }
    ]
  },
  comments: {
    comment: [
      {
        id: 3,
        comment: 'you be goat oo',
        createdAt: '2018-11-21T14:00:11.810Z',
        updatedAt: '2018-11-21T14:00:11.810Z',
        articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
        userId: 2,
        User: {
          username: 'nelson',
          email: 'nelson.nnaji@andela.com',
          Profile: null
        },
        likes: [],
        Replies: []
      }
    ]
  },
  user: {
    isAuth: true,
    loading: false
  },
  params: {
    articleslug: 'article-slxug'
  }
};
