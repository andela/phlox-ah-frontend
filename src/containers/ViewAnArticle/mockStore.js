export const articleStore = {
  Articles: {
    loading: false,
    success: false,
    Article: {
      loading: true,
      success: false,
      failure: false,
      article: {
        title: 'new test article by udochukwu',
        body: 'Body here',
        description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. ',
        imgUrl: 'https://images.pexels.com/photos/788485/pexels-photo-788485.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        readTime: 1,
        createdAt: '2018-11-12T18:16:18.630Z',
        updatedAt: '2018-11-12T18:16:18.630Z',
        User: {
          username: 'nnaji',
          Profile: {
            profileImage: 'https://images.pexels.com/photos/1065084/pexels-photo-1065084.jpeg?cs=srgb&dl=beautiful-beauty-brown-eyes-1065084.jpg&fm=jpg'
          }
        },
        Tags: [
          {
            id: 1,
            name: 'tech',
            createdAt: '2018-09-22T23:00:00.000Z',
            updatedAt: '2018-09-22T23:00:00.000Z',
          },
          {
            id: 2,
            name: 'animals',
            createdAt: '2018-09-22T23:00:00.000Z',
            updatedAt: '2018-09-22T23:00:00.000Z',
          }
        ],
        likes: [],
      }
    }
  },
  params: {
    articleslug: 'article-slxug'
  },
};
