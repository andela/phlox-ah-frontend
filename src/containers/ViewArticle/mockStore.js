export const articleStore = {
  article: {
    loading: true,
    success: true,
    failure: false,
    article: {
      title: 'new test article by udochukwu',
      body: 'Body here',
      description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. ',
      imgUrl: 'https://images.pexels.com/photos/788485/pexels-photo-788485.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
      readTime: 1,
      createdAt: '2018-11-12T18:16:18.630Z',
      updatedAt: '2018-11-12T18:16:18.630Z',
      User: {
        username: 'nnaji',
        Profile: {
          profileImage: 'https://images.pexels.com/photos/1065084/pexels-photo-1065084.jpeg?cs=srgb&dl=beautiful-beauty-brown-eyes-1065084.jpg&fm=jpg'
        }
      },
      Tags: [
        {
          id: 1,
          name: 'tech',
          createdAt: '2018-09-22T23:00:00.000Z',
          updatedAt: '2018-09-22T23:00:00.000Z',
        },
        {
          id: 2,
          name: 'animals',
          createdAt: '2018-09-22T23:00:00.000Z',
          updatedAt: '2018-09-22T23:00:00.000Z',
        }
      ],
      likes: [],
    }
  },
  comments: {
    comment: [
      {
        id: 3,
        comment: 'you be goat oo',
        createdAt: '2018-11-21T14:00:11.810Z',
        updatedAt: '2018-11-21T14:00:11.810Z',
        articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
        userId: 2,
        User: {
          username: 'nelson',
          email: 'nelson.nnaji@andela.com',
          Profile: null
        },
        likes: [],
        Replies: []
      }
    ]
  },
  user: {
    isAuth: true,
    loading: false
  },
  params: {
    articleslug: 'article-slxug'
  },
  bookmark: {
    bookmarks: []
  },
  report: {
    report: {},
    error: null,
    message: '',
    success: false,
    loading: false
  }
};
