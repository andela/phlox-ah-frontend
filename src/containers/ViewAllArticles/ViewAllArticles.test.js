import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ViewAllArticles from './ViewAllArticles';
import { articleStore } from './mockStore';
import { asyncActions } from '../BasePath';
import {
  RATE_ARTICLE, GET_ALL_COMMENT, CREATE_COMMENT, LIKE_ARTICLE, DISLIKE_ARTICLE
} from '../../actionTypes';
import CommentReducer from '../../reducers/CommentReducer';
// import ArticleCard from '../../components/ArticleCard/ArticleCard';

const mockStore = configureMockStore([thunk]);
const store = mockStore(articleStore);

let component;
let myComponent;

describe('<ViewArticle/>', () => {
  test('renders the ArticlePage Container', () => {
    component = shallow(
      <Provider store={store}>
        <ViewAllArticles />
      </Provider>
    );
    myComponent = component.dive({ context: { store } }).dive();
    expect(component.exists()).toBe(true);
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
  it('should have a main container', () => {
    expect(myComponent.find('.site-container').exists()).toBe(true);
  });
  it('should have a sidebar', () => {
    expect(myComponent.find('.sidebar').exists()).toBe(true);
  });
  it('should have function showAllArticles', () => {
    expect(myComponent.instance().showAllArticles).toBeDefined();
  });
  it('should have getArticles prop function', () => {
    expect(myComponent.instance().props.getArticles).toBeDefined();
  });
  it('should have articles state as array', () => {
    expect(myComponent.instance().state.articles).toMatchObject(articleStore.article.articles);
  });
});
