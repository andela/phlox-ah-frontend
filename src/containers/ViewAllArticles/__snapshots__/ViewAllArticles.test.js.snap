// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewAllArticles/> should render without throwing an error 1`] = `<Connect(ViewAllArticles) />`;
